src/llvm_ir_generator.cpp

```diff
@@ -0,0 +1,807 @@
+                  // Accessible only within this translation unit; declaring as
+                  // a data member introduces unnecessary dependency.
+    = PrevValueRecorder{};
+
+struct LabelViewInfo {
+  llvm::BasicBlock* entry;
+  llvm::BasicBlock* exit;
+  /// @brief This vector stores every `case` and `default` basic blocks of
+  /// a switch case.
+  /// This first element of a pair is the expression value
+  /// of a case statement.
+  /// This second element of a pair is the label's basic block.
+  std::vector<std::pair<llvm::Value*, llvm::BasicBlock*>> cases{};
+};
+
+/// @note Blocks that allows jumping within or out of it should add its labels
+/// to this list.
+auto
+    label_views_of_jumpable_blocks  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+    = std::vector<LabelViewInfo>{};
+
+}  // namespace
+
+void LLVMIRGenerator::Visit(const DeclStmtNode& decl_stmt) {
+  for (const auto& decl : decl_stmt.decls) {
+    decl->Accept(*this);
+  }
+}
+
+void LLVMIRGenerator::Visit(const VarDeclNode& decl) {
+  auto var_type = llvm_util_.GetLLVMType(*(decl.type));
+  // For function pointer, we need to change from FunctionType to PointerType
+  var_type = var_type->isFunctionTy() ? var_type->getPointerTo() : var_type;
+  auto addr = builder_->CreateAlloca(var_type);
+  if (decl.init) {
+    decl.init->Accept(*this);
+    auto val = val_recorder.ValOfPrevExpr();
+    builder_->CreateStore(val, addr);
+  }
+  id_to_val[decl.id] = addr;
+}
+
+void LLVMIRGenerator::Visit(const ArrDeclNode& arr_decl) {
+  auto arr_type = llvm_util_.GetLLVMType(*(arr_decl.type));
+  auto base_addr = builder_->CreateAlloca(arr_type, nullptr);
+  id_to_val[arr_decl.id] = base_addr;
+
+  auto arr_decl_type = dynamic_cast<ArrType*>(arr_decl.type.get());
+  for (auto i = std::size_t{0}, e = arr_decl_type->len(); i < e; ++i) {
+    if (i < arr_decl.init_list.size()) {
+      auto& arr_init = arr_decl.init_list.at(i);
+      arr_init->Accept(*this);
+    }
+
+    auto res_addr =
+        builder_->CreateConstInBoundsGEP2_32(arr_type, base_addr, 0, i);
+
+    if (i < arr_decl.init_list.size()) {
+      auto init_val = val_recorder.ValOfPrevExpr();
+      builder_->CreateStore(init_val, res_addr);
+    } else {
+      // set remaining elements as 0
+      auto zero = llvm::ConstantInt::get(llvm_util_.IntType(), 0, true);
+      builder_->CreateStore(zero, res_addr);
+    }
+  }
+}
+
+void LLVMIRGenerator::Visit(const RecordDeclNode& record_decl) {
+  /* Do nothing because this node only declares a type. */
+}
+
+void LLVMIRGenerator::Visit(const FieldNode& field) {
+  /* Do nothing because this node only declares a member type in a record. */
+}
+
+void LLVMIRGenerator::Visit(const RecordVarDeclNode& record_var_decl) {
+  auto* record_type = dynamic_cast<RecordType*>(record_var_decl.type.get());
+  assert(record_type);
+  auto type = llvm_util_.GetLLVMType(*(record_var_decl.type));
+  auto base_addr = builder_->CreateAlloca(type, nullptr);
```

Comment:
The default value is already `nullptr`. We can omit it.
```suggestion
  auto base_addr = builder_->CreateAlloca(type);
```